Description: Patches to correctly include standard C header files.
 Brahms has been compiled by the developer on a less picky toolchain
 gcc 4.7 on Debian 7 requires that correct C headers are #included.
Forwarded: By email to developer Ben Michinson
Author: Seb James <seb.james@sheffield.ac.uk>
Last-Update: 2014-02-20
--- a/components/test/busy.cpp
+++ b/components/test/busy.cpp
@@ -47,6 +47,7 @@
 #endif
 
 #ifdef __NIX__
+#include <unistd.h>
 #define msleep(msec) usleep(msec*1000)
 #endif
 
--- a/components/test/hang.cpp
+++ b/components/test/hang.cpp
@@ -39,6 +39,8 @@
 #define COMPONENT_CLASS_CPP client_brahms_test_hang_0
 #define COMPONENT_FLAGS F_NOT_RATE_CHANGER
 
+#include <unistd.h>
+
 //	include common header
 #include "../process.h"
 
--- a/components/test/sequence.cpp
+++ b/components/test/sequence.cpp
@@ -39,6 +39,8 @@
 #define COMPONENT_CLASS_CPP client_brahms_test_sequence_0
 #define COMPONENT_FLAGS (0)
 
+#include <unistd.h>
+
 //	include common header
 #include "../process.h"
 
--- a/framework/channel/channel.cpp
+++ b/framework/channel/channel.cpp
@@ -54,6 +54,9 @@
 #include "base/base.h"
 #include "channel.h"
 #include <queue>
+#ifdef __NIX__
+#include <unistd.h>
+#endif
 
 //	import namespaces
 using namespace brahms::channel;
--- a/framework/engine/base/os.cpp
+++ b/framework/engine/base/os.cpp
@@ -35,6 +35,7 @@
 
 #include <sys/time.h>
 #include <errno.h>
+#include <unistd.h>
 
 #endif
                             
--- a/framework/execute/os.cpp
+++ b/framework/execute/os.cpp
@@ -38,9 +38,9 @@
 
 
 #ifdef __NIX__
-
 #include <stdlib.h>
-
+#include <unistd.h>
+#include <sys/types.h>
 #endif
 
 
--- a/framework/execute/gui/gui.cpp
+++ b/framework/execute/gui/gui.cpp
@@ -68,7 +68,7 @@
 #ifdef __NIX__
 
 	#include <X11/Intrinsic.h>
-
+	#include <unistd.h>
 #endif
 
 
